### PR TITLE
MNT: Remove sphinx-asdf dependency

### PR DESCRIPTION
**Pull request by pllim**
Undo part of #8526 due to performance issues reported by @saimn at spacetelescope/sphinx-asdf#14 . Ideally, this should be backported to 3.2, if possible.

TODO:
- [x] See if we can [literalinclude](https://www.sphinx-doc.org/en/1.5/markup/code.html#includes) file YAML schemas from where they are (might be a problem if Sphinx does not know how to do that for files not in `docs` directory).
- [x] Let the whole CI run without skipping anything, just in case.

### CODE REVIEW

**Comment by pllim** [comment]
Note: The two CircleCI `mpl` failures were both socket timeouts from `visualization`, which I don't think is related to this PR.

File: docs/io/misc.rst [hunk 1]
```diff
@@ -12,9 +12,160 @@ functionality via the :class:`~astropy.table.Table` class itself (see
 listed in the `astropy.io.misc` section.
 
 .. automodapi:: astropy.io.misc
+   :headings: =-
 
 .. automodapi:: astropy.io.misc.hdf5
+   :headings: =-
 
 .. automodapi:: astropy.io.misc.yaml
+   :headings: =-
```
Comment (pllim) [inline docs/io/misc.rst]: Magic!

**Comment by pllim** [comment]
@saimn , I think this is great. Thank you!

https://30428-2081289-gh.circle-artifacts.com/0/home/circleci/project/docs/_build/html/io/misc.html#schemas

**Comment by saimn** [comment]
Yes, I think it's a good compromise, it makes things much simpler (and faster!).

**Pull request merged**